src/dialects/abstract/query-interface.js

```diff
@@ -641,8 +648,8 @@ export class QueryInterface {
    * Get foreign key references details for the table
    *
    * Those details contains constraintSchema, constraintName, constraintCatalog
-   * tableCatalog, tableSchema, tableName, columnName,
-   * referencedTableCatalog, referencedTableCatalog, referencedTableSchema, referencedTableName, referencedColumnName.
+   * tableCatalog, tableSchema, tableName, tableColumnNames,
```

Comment:
columns can only be in tables, I think this could be named `columnNames` and `referencedColumnNames`

Reply:
I respectfully challenge :) I gave this some careful consideration (because it'd be a chore to update).  I wish we had a Slack channel (other than `sequelize-help`) where this kind of stuff can be discussed before begin worked on.

It's important to point out there are three changes taking place here:
1. the names are pluralized to accommodate an array (for composite keys)
2. `table` is added to both `columnNames` columns
3. `table` is added to all targeted table parts (it pre-existed on _catalog_ and _schema_)

My conclusions:
1. It makes sense to logically group the elements with a word (or absence of a word) to distinguish the "from" and "to" connections. 
   - The other objects were all qualified with "table", "referenced", or "constraint", leaving certain elements floating and not part of those buckets.  By not qualifying `columnName`, it makes it slightly confusing because it does not follow the pattern.
   - I'd argue if `table` is dropped from `tableColumnNames`, then it should be dropped from `tableCatalog` and `tableSchema` as well for consistency
1. A bit of preference/opinion, but when sorting/grouping it is much nicer to have all the parts sorted alphabetically and grouped together.  To reiterate a former point, this makes it easier to read and work with the code

Sidenotes:
1. I also wanted to change "referenced" to present tense, but that was going too far 😆
1. My understanding is this doesn't translate into the ORM and only used internally (or as JSON results), as an ORM property, I would add the `columnNames` alias to point to `tableColumnNames`

Reply:
> the names are pluralized to accommodate an array

This change I'm 100% on board with, pluralize away

> The other objects were all qualified with "table", "referenced", or "constraint"

Not fully true, it's prefixed with `referenced` if it's the target element, and not prefixed otherwise (`tableCatalog` vs `referencedTableCatalog`, `tableSchema` vs `referencedTableSchema`, `tableName` vs `referencedTableName`, `columnName` vs `referencedColumnName`): it's always the exact same name + 'referenced'

I think for this PR, the change should be limited to only pluralizing the relevant names, we can look into having better names in a follow-up PR.

I would propose:

```
sourceCatalog
sourceSchema
sourceTable
sourceColumns

&

referencedCatalog
referencedSchema
referencedTable
referencedColumns
```

> only used internally

`queryInterface` is a public API (with some design issues), so API changes here must be considered breaking changes

Reply:
> Not fully true, it's prefixed with referenced if it's the target element, and not prefixed otherwise (tableCatalog vs referencedTableCatalog, tableSchema vs referencedTableSchema, tableName vs referencedTableName, columnName vs referencedColumnName): it's always the exact same name + 'referenced'

Yes, but the `referenced` prefix is a qualifier and the scheme is not immediately evident because lacking a qualifier could mean they're related or unrelated.  I like your proposal.  This is going to require a lot of updating to revert.  Is it possible that I can just move towards your proposal instead?  It might be less work to attempt to get buy-in on that.

> considered breaking changes
I noted that in one of the git commits, so I don't know what I was referring to with _this_.  However, if this is used through a property, I would again recommend aliasing the property to the corresponding column:
```
const fk = await getFirstFk();  // pseduo-code
fk.columns                      // populated by tableColumns or sourceColumns
```

Reply:
> Is it possible that I can just move towards your proposal instead?

I would very much prefer to do that in a separate PR as this one is growing too much in scope. Adding too many things to one PR makes it difficult to review and keep track of the breaking changes for our upgrade guide. You can have more than one PR open at a time if you want though